### PR TITLE
Add fish shell compatible output

### PR DESCRIPTION
**Pull request by chksome**
This adds a new argument `-shell` and the option to print fish shell compatible output. The default is still bash/zsh compatible output.

Documentation updated to match.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,38 +11,52 @@ Put `wsl2-ssh-agent` binary in your favorite directory in WSL2, for example, `$H
...
 ```
 chmod 755 wsl2-ssh-agent
 ```
 
 **If you are using ArchLinux, you can install the [wsl2-ssh-agent](https://aur.archlinux.org/packages/wsl2-ssh-agent) package from the AUR (maintained by @Hill-98).**
 
-### 2. Modify `.bashrc` (or `.zshrc` if you are using `zsh`)
+### 2. Modify your shell's rc file
 
+Bash or Zsh
```
Comment (mame) [inline README.md:31]:
```suggestion
#### Bash or Zsh
```

File: README.md [hunk 2]
```diff
@@ -11,38 +11,52 @@ Put `wsl2-ssh-agent` binary in your favorite directory in WSL2, for example, `$H
...
 Add the following line to `.bashrc` (or `.zshrc` if you are using `zsh`).
 
 ```
 eval $($HOME/wsl2-ssh-agent)
 ```
 
+Fish
```
Comment (mame) [inline README.md:38]:
```suggestion
#### Fish
```

File: README.md [hunk 3]
```diff
@@ -11,38 +11,52 @@ Put `wsl2-ssh-agent` binary in your favorite directory in WSL2, for example, `$H
...
+Add the following lines to config.fish
+
+```
+if status is-login
+  $HOME/wsl2-ssh-agent | source
```
Comment (mame) [inline README.md:43]: Shouldn't it be `$HOME/wsl2-ssh-agent --shell fish | source` ?

File: README.md [hunk 4]
```diff
@@ -11,38 +11,52 @@ Put `wsl2-ssh-agent` binary in your favorite directory in WSL2, for example, `$H
...
 Close and reopen the terminal and execute `ssh your-machine`.
 The command should communicate with ssh-agent.exe service.
 
 ## Troubleshooting
 
 ### Make sure that ssh-agent.exe is working properly
 
-* Open the "Services" app and check that "OpenSSH Authentication Agent" service is installed and running.
-* Check that `ssh your-machine` works perfect on cmd.exe or PowerShell, not on WSL2.
+- Open the "Services" app and check that "OpenSSH Authentication Agent" service is installed and running.
```
Comment (mame) [inline README.md:54]: If there is no particular reason, please do not change these.

**Review by mame — COMMENTED** [review]
Thanks. I think it is a good feature, but I am unsure if `--shell fish` is a good option. Do you know of any other examples?

I checked how `ssh-agent` does.

* `ssh-agent -s` outputs Bourne shell commands: `SSH_AUTH_SOCK=/path/to/sock; export SSH_AUTH_SOCK;`.
* `ssh-agent -c` outputs a C-shell command: `setenv SSH_AUTH_SOCK /path/to/sock`.
* If not specified, `ssh-agent` automatically checks if the environment variable `SHELL` ends with "csh".

And fish users seem to use `eval (ssh-agent -c)`. 

If we follow `ssh-agent`, it might be good to print `setenv` when `--csh` is given.

**Comment by mame** [comment]
I have reconsidered. I don't think it's a good idea to include the `--csh` option in modern times. If you show me that there are already major commands that implements `--shell fish` option, I'll be OK.

**Comment by Hill-98** [comment]
Maybe this is a good choice?

```bash
command -v export &>/dev/null && export A=1 || set A 1;:
```

File: README.md (hunk 3 above)
Comment (chksome) [reply README.md:43]: Yup, good catch. Will fix.

**Comment by chksome** [comment]
> I have reconsidered. I don't think it's a good idea to include the `--csh` option in modern times. If you show me that there are already major commands that implements `--shell fish` option, I'll be OK.

I agree that a switch for each shell seems antiquated. I was going to do that first then changed to the current `-shell` argument so the argument list doesn't get really polluted. I'll be honest though, I don't know of any other application with a `-shell` argument. rbenv handles this with a [shell name as an argument](https://github.com/rbenv/rbenv#basic-git-checkout) to `init`. Starship [does the same thing](https://starship.rs/#fish). What's here now is in the same spirit, but of course with a different name. I'm open to changing it as I just kind of picked the second thing that came to mind.

**Comment by mame** [comment]
How about this.

* Add `--format {auto,bash,zsh,csh,fish}`
* Default is `--format auto`, which changes the output depending on the SHELL environment variable
  * If `SHELL` ends with `fish`, print `set -x SSH_AUTH_SOCK /path/to/sock`
  * if `SHELL` ends with `csh`, print `setenv SSH_AUTH_SOCK /path/to/sock`
  * Otherwise, print `SSH_AUTH_SOCK=/path/to/sock; export SSH_AUTH_SOCK;`

Just in case, I want to leave room for future extensions, such as `--format bare-path` to show just the path as @Hill-98 said, or `--format custom:"export SSH_AUTH_SOCK=%s"` to use a custom format.

@Hill-98 I like polyglot too, but I am afraid that ordinary users might be suspicious when they see the output.

**Comment by mame** [comment]
@chksome I have created another pull request #10 with you as Co-Author. If you agree with it, I will merge it.

**Comment by chksome** [comment]
Better solution in #11. Closing this one.

**Pull request closed without merging**